src/data/openthaigpt_pretraining_data/mc4/preprocess.py

```diff
@@ -0,0 +1,943 @@
+script_words = ['function','var','click','margin','width','height','return','else','alert','<br>','href']
+
+#### Garbage Clean Words
+garbage_words = ['โหงวเฮ้ง','ครีมฟอกสี','ครีมผิวขาว','ฟอกสี','ไวท์เทนนิ่งครีม','ครีมไวท์เทนนิ่ง','ครีมลบฝ้ากระ','รับสร้างบ้าน','ครีมโรคสะเก็ดเงิน','บริการจองตั๋ว','บริการรีดผ้า','อาหารเสริมลดน้ำหนัก','ยาลดน้ำหนัก','ลดไขมัน','ผิงโซดา','สร้างบ้าน','ช่างกุญแจ','ช่างโลหะ','ช่างโยธา','ช่างเครื่องยนต์','ช่างไม้','ช่างกลโรงงาน','ช่างไฟฟ้า','ปรสิต','หนอน','เวิร์ม']
+
+#### Football teams
+football_teams = ['ยูเวนตุส','อินเตอร์ มิลาน','นาโปลี','เอซี มิลาน','ลาซิโอ','โรม่า','กัลโซ่','เอฟเวอร์ตัน','ซันเดอร์แลนด์','ลิเวอร์พูล','แมนเชสเตอร์','นิวคาสเซิล','เชลซี','อาร์เซนอล','คลิสตัลพาเลช','เซาแทมป์ตัน','เซาแธมป์ตัน','ไฮไลท์ฟุตบอล']
+
+#### Hotels Advertising
+hotel_ad = ['โรงแรมอันดับ','ที่พักแบบพิเศษอันดับ','สถานที่พักอันดับ','สถานที่พักคุ้มค่าอันดับ','โรงแรมใกล้กับ','โรงแรมที่ใกล้']
+
+#########
+# PRE-COMPILE REGEX to object for speed up processing.
+#########
+#-----------------------------------------------------
+# Remove useless row that make overhead in regex processing
+
+# Unusual row - line size too large
+# if there are 3 large lines ( 1500 characters each)
+toolarge_line_pattern = '.{1500}'
+toolarge_re = re.compile(toolarge_line_pattern, re.MULTILINE)
+
+sidebar_pattern = '.{0,40}(?:(?:\[|\()\d{0,9}(?:\]|\))(?:[ ]{0,2})?,?)'
+sidebar_re = re.compile(sidebar_pattern, re.MULTILINE)
+
+nonechar_pattern = '๮|๞|๨|๡|๷|๻|๫'
+nonechar_re = re.compile(nonechar_pattern, re.MULTILINE)
+
+none_tone_mark_pattern = 'ก าหนด|เป าหมาย|พ ฒนา|ค ณภาพ|ว จ ย|ค ณล กษณะ|ต างๆ|เป น |ให |บร หาร|ปร บปร ง|ใหม|อย าง|เง น'
+none_tone_mark_re = re.compile(none_tone_mark_pattern, re.MULTILINE)
+
+#-----------------------------------------------------
+
+gamble_pattern = '|'.join(gamble_words)
+gamble_re = re.compile(gamble_pattern, re.MULTILINE)
+
+football_pattern = '|'.join(football_teams)
+football_re = re.compile(football_pattern, re.MULTILINE)
+
+hotel_ad_pattern = '|'.join(hotel_ad)
+hotel_ad_re = re.compile(hotel_ad_pattern, re.MULTILINE)
+
+sale_pattern = '|'.join(sale_words)
+sale_re = re.compile(sale_pattern, re.MULTILINE)
+
+rent_pattern = '|'.join(rent_words)
+rent_re = re.compile(rent_pattern, re.MULTILINE)
+
+json_pattern = "\s*\"(?:\w)*\"\s*:"
+json_re = re.compile(json_pattern, re.MULTILINE)
+
+script_pattern = r'\b' + '|'.join(script_words) + r'\b'
+script_re = re.compile(script_pattern, re.MULTILINE)
+
+garbage_pattern = '|'.join(garbage_words)
+garbage_re = re.compile(garbage_pattern, re.MULTILINE)
+
+ghost_pattern = 'เธฃเน|เธเธญ|เธเน|เธฐเธ|เธฅเธฐ|เธซเธฒ|เธญเธฒ|เธดเธ|เธตเธข|เธญเน|เธญเธ|เธดเน|เธฑเธ|เธกเน|เธฒเธ|เธชเน|เน€เธ'
+ghost_re = re.compile(ghost_pattern, re.MULTILINE)
+
+url_pattern = '\\b(?:(?:https?|ftp)://[^\s/$\.\?#].[^\s]*)\\b|\\b(?:www\.?)?(?:(?:[\w-]*)\.)*(?:com|net|org|info|biz|me|io|co|asia|xyz|th|cn|in|uk|jp|ru)\\b'
+url_re = re.compile(url_pattern, re.MULTILINE)
+
+menu1_pattern = '\|(?:[^\|\n]*\|)+.*'
```

Comment:
Menu patterns are one of dangerous for mC4 data.  Because, mC4 data has some huge strange repeat menu pattern in a row.  Those data patterns reduce the performance of regex too much.  So I split it off.

Reply:
I try to merge it by add non-capturing group `(?:)` and join it using `|`. It seems to work well in my opinion but im not that sure. If you have time please check this out.

https://colab.research.google.com/drive/1Xzwp3NN1cDu6zTzjXpNPoU_FVPvQgODk#scrollTo=Th__C3xXGOsI

If it really have to run 4 patterns separately, I suggest that we list format might be easier to read.

```
menu_pattern = [menu1_pattern,......]

menu_re = [re.compile(elem) for elem in menu_pattern 

```

Reply:
The result is different.  menu_re will do 4 patterns in one menu_re.sub().  But as I designed ,  I want to match each pattern with replace blank in sequence.   This is a one of difference  in logic as I designed patterns in sequence too.  Krub.

Reply:
I'm sorry for the menu_patterns.  The menu patterns can do as you suggest because they are separate patterns.   I will do as you suggest.  But many patterns can't do like this.  

Thank you, krub.    
Note:  menu patterns is one of a big problem if we not clean large menu pattern out when read a row.   It is very huge one line (\n) data.  But I clean them at start of process.